zebra/rt_netlink.c

```diff
@@ -813,6 +813,7 @@ static int netlink_route_change_read_unicast(struct nlmsghdr *h, ns_id_t ns_id,
 					nh.bh_type = bh_type;
 				}
 				nh.ifindex = index;
+				nh.vrf_id = vrf_id;
```

Comment:
The index( nh.ifindex ) should dictate the nh's vrf.  This will break in a route leaking situation.

Reply:
ahh I see what you mean.

yes this code just needs to be changed to call `parse_nexthop_unicast()`

Reply:
That handles the vrf_id by looking up the ifindex and using the interface's if it finds it, but falls back to the one set here if it doesn't.